src/python/grpcio/grpc/_cython/_cygrpc/time.pyx.pxi

```diff
@@ -15,10 +15,9 @@
 cdef gpr_timespec _timespec_from_time(object time):
   return (gpr_inf_future(GPR_CLOCK_REALTIME)
           if time is None else
-          gpr_time_from_nanos(<int64_t>(time * 1e9), GPR_CLOCK_REALTIME))
+          gpr_time_from_millis(<int64_t>(int(float(time) * 1000)), GPR_CLOCK_REALTIME)
```

Comment:
Why do you consider this version "safer"?

Reply:
The clock-type mismatch is caused by the OverflowError. This version still causing test case to fail in Windows:

```Cython
gpr_time_from_nanos(<int64_t>(time * 1e9), GPR_CLOCK_REALTIME)
```

That one are just proposed fix... To see if the Windows test are passed.

I guess Cython is converting Python `int` into Cpp `long` by default. The root cause is that the length of `long` is shorter in Windows.
https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=vs-2017

What I can do now is trying various ways to make Cython to covert a Python `int` directly into `int64_t`.

Reply:
By the way, feel free to remove my commits since they fixed nothing.

Also, welcome trying other workarounds for this problem.

Reply:
It's hard for me to experiment without Windows on hand, but can you please explain what does Cython try to convert to `int64_t`?

Reply:
I'm not 100% percent sure about hat, what other situation can cause OverflowError?

Reply:
One way to find out would be to peek into the `cygrpc.cpp` source file generated.